### ISSUE TITLE
Bug: mock that return set of Mocks fails

### ISSUE THREAD

**Issue by alaxwork** [mockk/mockk#221]
### Description

I trying to mock the following interfaces: 
```kotlin
interface Foo {
    fun getTasks(): Set<Task>
}

interface Task {
    fun doIt()
}
```
Test example:
```kotlin
@Test
fun prepare() {
    val foo = mockk<Foo>()
    every { foo.getTasks() } returns setOf(mockk(), mockk())
}
```
Special for the test that mock returns set of mocks again. HashMap invokes then hashCode method of the mock object.

### Expected Behavior

The test should pass

### Current Behavior

The test fails

#### Stack trace

```
io.mockk.MockKException: Bad recording sequence. Please finalize every { ... } block with returns/answers/just Runs

	at io.mockk.impl.recording.states.CallRecordingState.cancelAndThrowBadRecordingState(CallRecordingState.kt:30)
	at io.mockk.impl.recording.states.CallRecordingState.call(CallRecordingState.kt:12)
	at io.mockk.impl.recording.CommonCallRecorder.call(CommonCallRecorder.kt:54)
	at io.mockk.impl.stub.MockKStub.handleInvocation(MockKStub.kt:272)
	at io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation(JvmMockFactoryHelper.kt:25)
	at io.mockk.proxy.jvm.advice.Interceptor.call(Interceptor.kt:20)
	at io.mockk.proxy.jvm.advice.BaseAdvice.handle(BaseAdvice.kt:42)
	at io.mockk.proxy.jvm.advice.jvm.JvmMockKProxyInterceptor.intercept(JvmMockKProxyInterceptor.java:31)
	at Task$Subclass2.hashCode(Unknown Source)
	at java.util.HashMap.hash(HashMap.java:338)
	at java.util.HashMap.put(HashMap.java:611)
	at java.util.HashSet.add(HashSet.java:219)
	at kotlin.collections.ArraysKt___ArraysKt.toCollection(_Arrays.kt:7522)
	at kotlin.collections.ArraysKt___ArraysKt.toSet(_Arrays.kt:7857)
	at kotlin.collections.SetsKt__SetsKt.setOf(Sets.kt:42)
	at Test.prepare(Test.kt:18)
```

#### Minimal reproducible code (the gist of this issue)

```kotlin
import io.mockk.every
import io.mockk.mockk
import org.junit.jupiter.api.Test

interface Foo {
    fun getTasks(): Set<Task>
}

interface Task {
    fun doIt()
}

internal class Test {

    @Test
    fun prepare() {
        val foo = mockk<Foo>()
        every { foo.getTasks() } returns setOf(mockk(), mockk())
    }
}
```

**Comment by oleksiyp** [comment]
Looks seriius, thanks for the report

**Comment by jcornaz** [comment]
Isn't the same issue as #201 ?

Currently MockK doesn't support creating mocks during stubbing of another mock.

The following version passes, since the mock are created *before* the stubbing start.
```kotlin
  @Test
  fun prepare() {
    val foo = mockk<Foo>()
    val set = setOf(mockk<Task>(), mockk<Task>())
    every { foo.getTasks() } returns set
  }
```

**Comment by oleksiyp** [comment]
@jcornaz missed this point. It might be, but I have doubts. There we have stubbing inside of unfinished stubbing.

**Comment by oleksiyp** [comment]
Released [1.9.1](https://github.com/mockk/mockk/releases/tag/1.9.1). Please try it out and close the ticket if everything is fine

**Comment by alaxwork** [comment]
I've tested. It's working as expected. Thank you.
